### ISSUE TITLE
[CI] ArchiveTests and CertGenCliTests failing on Windows

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#50825]
Our `ArchiveTests.test50StartAndStop` tests are [failing consistently](https://gradle-enterprise.elastic.co/scans/tests?failures.failureClassification=non_verification&list.offset=0&list.size=50&list.sortColumn=startTime&list.sortOrder=desc&search.buildToolType=gradle&search.buildToolType=maven&search.startTimeMax=1578610235297&search.startTimeMin=1576137600000&search.tags=ci&search.tags=windows&tests.container=org.elasticsearch.packaging.test.ArchiveTests&tests.sortField=FAILED&tests.test=test50StartAndStop&tests.unstableOnly&trends.section=overview&trends.timeResolution=day&viewer.tzOffset=-480) in `7.x` and `7.5` due to a failure to delete some temporary files. Interestingly the `DebMetadataTests` are [failing for a similar reason](https://gradle-enterprise.elastic.co/scans/tests?failures.failureClassification=non_verification&list.offset=0&list.size=50&list.sortColumn=startTime&list.sortOrder=desc&search.buildToolType=gradle&search.buildToolType=maven&search.startTimeMax=1578610235297&search.startTimeMin=1576137600000&search.tags=ci&search.tags=windows&tests.container=org.elasticsearch.packaging.test.DebMetadataTests&tests.sortField=FAILED&tests.test=classMethod&tests.unstableOnly&trends.section=overview&trends.timeResolution=day&viewer.tzOffset=-480) which I've opened a PR to deal with (https://github.com/elastic/elasticsearch/pull/50824) in a different way.

What I find interesting is that we are _not_ having this issue in `master` so have we done something different there with regards to deleting files and cleanup to workaround Windows file lock issues? @rjernst @williamrandolph thoughts?

**Comment by mark-vieira** [comment]
FYI, to put in perspective how impactful this issue is, of the 1740 builds in the past month that have failed due to test failures, 543 of them are due to this issue. That's over 30%. We should give this high priority given it's consistent and accounts for such a high percentage of our build failures.

**Comment by mark-vieira** [comment]
I'm just going to reopen this until we confirm that #54884 solves this issue.

**Comment by mark-vieira** [comment]
@rjernst Looks like we are still failing here. It really seems like that process is still running. Check out the error:

```
[2020-01-14T12:44:40,245][INFO ][o.e.p.u.Shell            ] [TEST-ArchiveTests.test50StartAndStop-seed#[E0C38CBC13FB5E05]-workerest_thread_info] Ran: [powershell.exe, -Command, $processInfo = New-Object System.Diagnostics.ProcessStartInfo; $processInfo.FileName = 'C:\tmp\elasticsearch\bin\elasticsearch.bat'; $processInfo.Arguments = '-p C:\tmp\elasticsearch\elasticsearch.pid'; $processInfo.RedirectStandardOutput = $true; $processInfo.RedirectStandardError = $true; $processInfo.RedirectStandardInput = $true; $processInfo.Environment.Add('JAVA_HOME', 'C:\Users\jenkins\.java\java8'); $processInfo.UseShellExecute = $false; $process = New-Object System.Diagnostics.Process; $process.StartInfo = $processInfo; $outScript = { $EventArgs.Data | Out-File -Encoding UTF8 -Append 'C:\tmp\elasticsearch\logs\output.out' }; $errScript = { $EventArgs.Data | Out-File -Encoding UTF8 -Append 'C:\tmp\elasticsearch\logs\output.err' }; $stdOutEvent = Register-ObjectEvent -InputObject $process -Action $outScript -EventName 'OutputDataReceived'; $stdErrEvent = Register-ObjectEvent -InputObject $process -Action $errScript -EventName 'ErrorDataReceived'; $process.Start() | Out-Null; $process.BeginOutputReadLine(); $process.BeginErrorReadLine(); Wait-Process -Timeout 10 -Id $process.Id; $process.Id;] exitCode = [0] stdout = [5772] stderr = [Wait-Process : This command stopped operation because process "cmd (5772)" is not stopped in the specified time-out.
At line:1 char:1061
+ ... rorReadLine(); Wait-Process -Timeout 10 -Id $process.Id; $process.Id;
+                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (System.Diagnostics.Process (cmd):Process) [Wait-Process], TimeoutException
    + FullyQualifiedErrorId : ProcessNotTerminated,Microsoft.PowerShell.Commands.WaitProcessCommand]
```

So the powershell cmd to wait for process termination is timing out. Should we increase that timeout beyond 10 seconds?

**Comment by rjernst** [comment]
The file being held onto is not created by Elasticsearch, it is created from Gradle. I will investigate more.

**Comment by mark-vieira** [comment]
These tests (along with `CertGenCliTests` account for about 50 build failures every day. If we are waiting on something for infra to provide to us to sort these out then we need to mute these in the meantime.

**Comment by ywangd** [comment]
A couple failures from today:
* https://gradle-enterprise.elastic.co/s/i6ara3nlys5te
* https://gradle-enterprise.elastic.co/s/k2wkhu7ff5rwo
* https://gradle-enterprise.elastic.co/s/ov3kb2mod6nm2

**Comment by rjernst** [comment]
closed by #58010
